tests/TestCase/Http/Middleware/HttpsEnforcerMiddlewareTest.php

```diff
@@ -0,0 +1,97 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @since         4.0.0
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Test\TestCase\Http\Middleware;
+
+use Cake\Http\Exception\BadRequestException;
+use Cake\Http\Middleware\HttpsEnforcerMiddleware;
+use Cake\Http\Response;
+use Cake\Http\ServerRequest;
+use Cake\TestSuite\TestCase;
+use TestApp\Http\TestRequestHandler;
+use Zend\Diactoros\Response\RedirectResponse;
+use Zend\Diactoros\Uri;
+
+/**
+ * Test for HttpsEnforcerMiddleware
+ */
+class HttpsEnforcerMiddlewareTest extends TestCase
+{
+    public function testForRequestWithHttp()
+    {
+        $uri = new Uri('https://localhost/foo');
+        $request = new ServerRequest();
+        $request = $request->withUri($uri);
+
+        $handler = new TestRequestHandler(function ($req) {
+            return new Response();
```

Comment:
Should you set something on this response and check that header/body content is the one returned?